src/Chainweb/Pact/PactService.hs

```diff
@@ -720,20 +707,29 @@ execNewBlock
     -> Miner
     -> BlockCreationTime
     -> PactServiceM cas PayloadWithOutputs
-execNewBlock mpAccess parentHeader miner creationTime = withDiscardedBatch $ do
-    withCheckpointerRewind (Just (bHeight, pHash)) "execNewBlock" $ \pdbenv -> do
+execNewBlock mpAccess parentHeader miner creationTime =
+  do
+    updateMP
```

Comment:
You moved it outside but I'm not sure there's a reason to, neither of the functions called should touch the DB

Reply:
This was just to make it clear that this doesn't "roll back" or something. When I see stuff bracketed I assume the scope of the update is somehow contained, so putting it outside makes it easier for me at least to understand.